ipld/blockstore/src/buffered.rs

```diff
@@ -7,17 +7,18 @@ use super::BlockStore;
 use byteorder::{BigEndian, ByteOrder, ReadBytesExt};
 use cid::{Cid, Code, DAG_CBOR};
 use db::{Error, Store};
+use std::borrow::Borrow;
 use std::collections::HashMap;
 use std::error::Error as StdError;
 use std::io::{Read, Seek};
-use std::{cell::RefCell, convert::TryFrom, io::Cursor};
+use std::{convert::TryFrom, io::Cursor};
 
 /// Wrapper around `BlockStore` to limit and have control over when values are written.
 /// This type is not threadsafe and can only be used in synchronous contexts.
 #[derive(Debug)]
 pub struct BufferedBlockStore<'bs, BS> {
     base: &'bs BS,
-    write: RefCell<HashMap<Cid, Vec<u8>>>,
+    write: std::sync::Mutex<HashMap<Cid, Vec<u8>>>,
```

Comment:
Any reason for not using a lock free map?

Reply:
Replaced with DashMap